Src/DependencyCollector/Shared/DependencyTrackingTelemetryModule.cs

```diff
@@ -237,7 +242,7 @@ private void InitializeForDiagnosticAndFrameworkEventSource()
                     this.SetComponentCorrelationHttpHeaders,
                     this.ExcludeComponentCorrelationHttpHeadersOnDomains,
                     this.EffectiveProfileQueryEndpoint);
-                this.httpDesktopDiagnosticSourceListener = new HttpDesktopDiagnosticSourceListener(desktopHttpProcessing);
+                this.httpDesktopDiagnosticSourceListener = new HttpDesktopDiagnosticSourceListener(desktopHttpProcessing, new ApplicationInsightsUrlFilter(this.telemetryConfiguration));
```

Comment:
this is supported by the AppInsights Url filer: it checks if endpoint was changed and caches new value if needed